### PR TITLE
[WIP] [9.0] Account financial report QWeb - General Ledger

### PR DESCRIPTION
**Pull request by yvaucher**
This PR intend to create an efficient report for the community version of Odoo
Plus replace the module `account_financial_report_webkit`

For now only GL has been made.

It take the exact same layout as `account_financial_report_qweb`

What is new is that it uses PSQL window function to compute the data of the report.
http://www.postgresql.org/docs/current/static/tutorial-window.html

### CODE REVIEW

**Comment by fclementic2c** [comment]
@hbto FYI :)

File: account_fiscalyear/models/account_fiscalyear.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8 -*-
+# © 2015 Yannick Vaucher (Camptocamp)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).-
+from openerp import models, fields, api, _, exceptions
+
+
+class AccountFiscalyear(models.Model):
+    _name = 'account.fiscalyear'
```
Comment (gurneyalex) [inline account_fiscalyear/models/account_fiscalyear.py:8]: Missing ACLs for this model

File: account_financial_report_qweb/wizard/general_ledger_wizard.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                          'journal_ids',
+                          'centralize',
+                          'target_move',
+                          'date_from',
+                          'date_to',
+                          'fiscalyear'])[0]
+        data['form'].update(vals)
+        return data
+
+    @api.onchange('filter')
```
Comment (gurneyalex) [inline account_financial_report_qweb/wizard/general_ledger_wizard.py:84]: filter is not a field of the model, unless I'm mistaken.

File: account_financial_report_qweb/__init__.py [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# -*- coding: utf-8 -*-
+# © 2015 Yannick Vaucher
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).-
+from . import models
```
Comment (elicoidal) [inline account_financial_report_qweb/__init__.py:4]: Hi @yvaucher
Please sort all imports in alphabetical order following: https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#imports

File: account_financial_report_qweb/__openerp__.py [hunk 4]
```diff
@@ -0,0 +1,190 @@
+# -*- encoding: utf-8 -*-
```
Comment (elicoidal) [inline account_financial_report_qweb/__openerp__.py:1]: Please use simple headers in all py files following OCA templates
https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L3

File: account_financial_report_qweb/__openerp__.py [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+{
+    'name': 'Financial Reports - Qweb',
+    'description': """
```
Comment (elicoidal) [inline account_financial_report_qweb/__openerp__.py:23]: Remove description => README

File: account_financial_report_qweb/__openerp__.py [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+* The report aggregate data per account with sub-totals
+* Initial balance must be calculated the same way that
+  the partner balance / Ignoring the opening entry
+  in special period (idem open invoice report)
+* Only accounts with internal type payable or receivable are considered
+  (idem open invoice report)
+* If maturity date is null then use move line date
+
+
+Limitations:
```
Comment (elicoidal) [inline account_financial_report_qweb/__openerp__.py:144]: issue/roadmap in README

File: account_financial_report_qweb/__openerp__.py [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+------------
+
+Totals for amounts in currencies are effective if the partner belongs to
+an account with a secondary currency.
+
+""",
+    'version': '8.0.1.1.0',
```
Comment (elicoidal) [inline account_financial_report_qweb/__openerp__.py:151]: version 9.0.1.0.0

File: account_financial_report_qweb/__openerp__.py [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+    'author': "Camptocamp,Odoo Community Association (OCA)",
+    'license': 'AGPL-3',
+    'category': 'Finance',
+    'website': 'http://www.camptocamp.com',
+    'images': [
+        'images/ledger.png', ],
+    'depends': ['account',
+                'account_fiscalyear'],
+    'demo': [],
+    'data': [#'account_view.xml',
```
Comment (elicoidal) [inline account_financial_report_qweb/__openerp__.py:161]: remove the comments

File: account_financial_report_qweb/models/account.py [hunk 9]
```diff
@@ -0,0 +1,15 @@
+# -*- coding: utf-8 -*-
+# © 2011 Guewen Baconnier (Camptocamp)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).-
+from openerp import models, fields
```
Comment (elicoidal) [inline account_financial_report_qweb/models/account.py:4]: order

File: account_financial_report_qweb/report/common_reports.py [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+    centralize = fields.Boolean()
+    target_move = fields.Char()
+
+    filter = fields.Selection(
+        [('filter_no', 'No Filters'),
+         ('filter_date', 'Date'),
+         ('filter_opening', 'Opening Only')],
+        "Filter by",
+        required=False,
+        help='Filter by date: no opening balance will be displayed. '
```
Comment (elicoidal) [inline account_financial_report_qweb/report/common_reports.py:77]: remove period

File: account_financial_report_qweb/report/common_reports.py [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+             '(opening balance can only be computed based on period to be '
```
Comment (elicoidal) [inline account_financial_report_qweb/report/common_reports.py:78]: "opening balance can only be computed if the provided period is correct."

File: account_financial_report_qweb/static/src/css/report.css [hunk 12]
```diff
@@ -0,0 +1,80 @@
+body, table, td, span, div {
+    font-family: Helvetica, Arial;
```
Comment (elicoidal) [inline account_financial_report_qweb/static/src/css/report.css:2]: are this font UTF8 compatible?

File: account_fiscalyear/__openerp__.py [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).-
+{
+    'name': 'Account fiscal year',
+    'summary': 'Add fiscal year',
+    'version': '9.0.1.0.0',
+    'author': "Camptocamp,Odoo Community Association (OCA)",
+    'license': 'AGPL-3',
+    'category': 'Finance',
+    'website': 'http://www.camptocamp.com',
+    'images': [],
```
Comment (elicoidal) [inline account_fiscalyear/__openerp__.py:12]: remove empty keys

File: account_financial_report_qweb/__init__.py (hunk 3 above)
Comment (yvaucher) [inline account_financial_report_qweb/__init__.py:4]: Damn, I have always set the same order as the order of use. :-/

models, fields, api

File: account_financial_report_qweb/__openerp__.py (hunk 8 above)
Comment (yvaucher) [inline account_financial_report_qweb/__openerp__.py:161]: It's WIP and some must be come back

**Comment by sbidoul** [comment]
@yvaucher could you do a separate PR with account_fiscal_year? It's easier to review and merge quickly and can serve as a basis for other things (eg we can use it in mis_builder too).

**Comment by sbidoul** [comment]
@yvaucher @fclementic2c actually I'm wondering if we should not simply have an `account_fiscal_period` module which simply provides the possibility to have pre-set reporting periods. Such periods could have any duration, and overlap so we could have in the same table fiscal month, quarters or years.

**Comment by luc-demeyer** [comment]
+1 for account_fiscal_period

**Comment by elicoidal** [comment]
fiscal periods are for filter but as well for control (cannot post in closed period).
+1 to have it in separate module

**Comment by sbidoul** [comment]
> as well for control (cannot post in closed period).

At this stage I don't think we need periods for control. 9.0 has `period_lock_date` and `fiscalyear_lock_date` out of the box and that should sufficient (with `account_move_locking`).
Missing is per journal lock date, but that is not related to periods.

**Comment by elicoidal** [comment]
@sbidoul actually my point was to force some control at month level, not fiscal year level.
Idea is not to be able to post anything for the month being closed. Period_lock_date will forbid any pos after certain date. I am personally fine with that but some countries might need to be able to lock previous period and fix the last before last.

**Comment by fclementic2c** [comment]
This PR has been superseeded by [(https://github.com/OCA/account-financial-reporting/pull/211)]

**Pull request closed without merging**